### ISSUE TITLE
STUN requested from WhatsApp are not properly passed (related to #174)

### ISSUE THREAD

**Issue by quaqo** [IngoZenz/personaldnsfilter#181]
I think this might be related to #174.

With pDNSf active (latest version), I can't make WhatsApp calls.
It's independent of the configuration (fresh install is the same) and of the Always on VPN setting.
Android version is 10.

After analisying the traffic with Wireshark (between the router and the Android device) I can see that the STUN UDP allocation requests have **10.0.2.15** as the source IP.

The STUN server is never able to reply back with a "Allocate success response" message.

If I close pDNSf, the STUN request correctly originate from the device IP on the network, and the call is initiated.

I ran Wireshark for a complete hour. And the only datagrams with the same source IP of the VPN are those STUN requests.

All other datagrams (but also TCP connection) originate as coming from the device IP.

Any ideas?

**Comment by IngoZenz** [comment]
cannot reproduce this here... any special config? Did you try with default config?
Feel free to post your dnsfilter.conf file.

**Comment by quaqo** [comment]
As I wrote:

> It's independent of the configuration (fresh install is the same) and of the Always on VPN setting.
> Android version is 10.

No special config. I reproduced it just by doing a fresh install of both pDNSf and WhatsApp on test phone with default dnsfilter.conf and with VPN Always Active off.

It's the same both with Mobile Data (no NAT, public IP) and WiFi (I have a few natted hops between the device and the Internet).

The only thing is pDNSf.

One thing I can try is to install an older version of Android on the test phone, since all devices I have available now are 10, unless you already tried with Android 10 and you couldn't reproduce it.

I'm using the latest F-Droid version of pDNSf with default config, as said above.

**Comment by quaqo** [comment]
@IngoZenz I wonder if [this](https://github.com/julian-klode/dns66/issues/193) might be related.

**Comment by IngoZenz** [comment]
yes - could be - but I would only expect issues in combination with routeIPs / routeUnderlyingDNS setting which is disabled in default config.
Anyhow I do not thing pDNSf can do much here...

**Comment by quaqo** [comment]
So. To add to this: I can't reproduce this with the very same configuration that I have on my phone (including routeIPs) on AVM in Android Studio.
